packages/commonwealth/client/scripts/views/modals/confirm_snapshot_vote_modal.tsx

```diff
@@ -44,6 +45,36 @@ export const ConfirmSnapshotVoteModal = (
 
   const [isSaving, setIsSaving] = React.useState<boolean>(false);
 
+  const { trackAnalytics } = useBrowserAnalyticsTrack({ onAction: true });
+
+  const handleVote = async (e) => {
+    e.preventDefault();
+    setIsSaving(true);
+    const votePayload = {
+      space: space.id,
+      proposal: id,
+      type: 'single-choice',
+      choice: parseInt(selectedChoice) + 1,
+      metadata: JSON.stringify({}),
+    };
+    try {
+      castVote(author.address, votePayload).then(async () => {
+        await app.snapshot.refreshProposals();
+        onModalClose();
+        successCallback();
+      });
+    } catch (err) {
+      console.log(err);
+      const errorMessage = err.message;
+      notifyError(errorMessage);
+    }
+    setIsSaving(false);
+
+    trackAnalytics({
+      event: MixpanelSnapshotEvents.SNAPSHOT_VOTE_OCCURRED,
+    });
```

Comment:
event will be fired regardless success or failure. Shouldn't it be only at the end of `try` cluase?